sphinx/ext/inheritance_diagram.py

```diff
@@ -297,9 +300,15 @@ def generate_dot(self, name: str, urls: Dict[str, str] = {},
         for name, fullname, bases, tooltip in sorted(self.class_info):
             # Write the node
             this_node_attrs = n_attrs.copy()
-            if fullname in urls:
-                this_node_attrs['URL'] = '"%s"' % urls[fullname]
-                this_node_attrs['target'] = '"_top"'
+
+            # Fuzzy matching in case the reference is at a different level of the namespace
+            fullname_ends = itemgetter(0, -1)(fullname.split('.'))
+            for refname, url in urls.items():
+                if itemgetter(0, -1)(refname.split('.')) == fullname_ends:
```

Comment:
```suggestion
            fullname_start, *_, fullname_end = fullname.split('.')
            for refname, url in urls.items():
                start, *_, end = (refname.split('.')
                if (fullname_start, fullname_end) == (start, end):
```

Though I'm still not really sure what this logic is for / what the change achieves--could you refactor further or add explanatory notes please?

Reply:
Yup, will do!  Once I finish the unit test, I should be able to verify my understanding.  In a nutshell, the issue is that a class can derive from an external class that is documented in a different location than its actual native location.  For example, say `internal_class` derives from `external_class`, where `external_class`'s code definition is in:
```
external_package.some_module.external_class
```
However, the external package can choose to import `external_class` one level up in the namespace:
```
external_package.external_class
```
and then document it only there.  That result is that the inventory has a listing for `external_package.external_class` but does not have a listing for `external_package.some_module.external_class`.  So, when creating an inheritance diagram for `internal_class`, `inheritance_diagram` will create a reference for `external_package.some_module.external_class`, and `intersphinx` will return a mapping that associates `external_package.external_class` with a URL.  Without fuzzy matching, `inheritance_diagram` sees that there is no `external_package.some_module.external_class` in the mapping, so does not create the desired link.

Reply:
Moot (explained below)